drivers/flash/soc_flash_nrf_rram.c

```diff
@@ -47,8 +47,7 @@ static struct k_sem sem_lock;
 #define WRITE_LINE_SIZE       16 /* In bytes, one line is 128 bits. */
 #define WRITE_BUFFER_MAX_SIZE (WRITE_BUFFER_SIZE * WRITE_LINE_SIZE)
 BUILD_ASSERT((PAGE_SIZE % (WRITE_LINE_SIZE) == 0), "erase-block-size must be a multiple of 16");
-BUILD_ASSERT((WRITE_BLOCK_SIZE_FROM_DT % (WRITE_LINE_SIZE) == 0),
-	     "if NRF_RRAM_WRITE_BUFFER_SIZE > 0, then write-block-size must be a multiple of 16");
```

Comment:
Shouldn't the check be whether `WRITE_BUFFER_SIZE` is multiple of `WRITE_LINE_SIZE` ?
In else shouldn't we have a check where whatever is set to `WRITE_BLOCK_SIZE_FROM_DT` is multiple of `WRITE_LINE_SIZE`, which is minimal alignment for the RRAM?

Reply:
I had discussed this change offline before opening the PR and as I understood it, RRAM can write single bytes unlike the RAM on other platforms, so it wasn't as applicable here. 
Based on the Kconfig description for `CONFIG_NRF_RRAM_WRITE_BUFFER_SIZE` I dont see the purpose of checking `WRITE_BUFFER_SIZE` is multiple of `WRITE_LINE_SIZE` since the `WRITE_BUFFER_SIZE` is the ["number of 128 bit words"](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/kconfig/index.html#!CONFIG_NRF_RRAM_WRITE_BUFFER_SIZE) while the `WRITE_LINE_SIZE` is the [size of each line](https://github.com/zephyrproject-rtos/zephyr/pull/69470/commits/8db41ace12ca9dfd9372efff3239b4dfbe152aa9#diff-fad767096349b533ba54e1e4c53c2df4019343bbbf45353aeba8b11f42a9ef1aR47), so the relationship between the line size and buffer size is just that the `buffer * line_size=total_buffer_size`